File: Backend/src/main/java/com/woovi/crudbank/balance/application/BalanceService.java
package com.woovi.crudbank.balance.application;

import com.woovi.crudbank.account.application.AccountService;
import com.woovi.crudbank.account.domain.AccountDocument;
import org.springframework.stereotype.Service;

import java.math.BigDecimal;

@Service
public class BalanceService {

    private final AccountService accountService;

    public BalanceService(AccountService accountService) {
        this.accountService = accountService;
    }

    public BigDecimal availableBalance(String accountGlobalId) {
        AccountDocument account = accountService.findRawByGlobalId(accountGlobalId);
        return account.getCurrentBalance();
    }
}

File: backend/src/modules/balance/balanceService.ts
import { AccountService } from '../account/accountService';

export class BalanceService {
  private readonly accountService: AccountService;

  constructor(accountService: AccountService) {
    this.accountService = accountService;
  }

  async availableBalance(accountGlobalId: string): Promise<string> {
    return this.accountService.getCurrentBalanceByGlobalId(accountGlobalId);
  }
}
